homeassistant/components/sensor/hp_ilo.py

```diff
@@ -121,23 +143,15 @@ def update(self):
         self.hp_ilo_data.update()
         ilo_data = getattr(self.hp_ilo_data.data, self._ilo_function)()
 
-        # Store the data received from the ILO API
-        if isinstance(ilo_data, dict):
-            self._data = ilo_data
-        else:
-            self._data = {'value': ilo_data}
+        if self._sensor_value_template is not None:
+            ilo_data = self._sensor_value_template.render(ilo_data=ilo_data)
 
         # If the data received is an integer or string, store it as
-        # the sensor state
-        if isinstance(ilo_data, (str, bytes)):
-            states = [STATE_ON, STATE_OFF]
-            try:
-                index_element = states.index(str(ilo_data).lower())
-                self._state = states[index_element]
-            except ValueError:
-                self._state = ilo_data
-        elif isinstance(ilo_data, (int, float)):
+        # the sensor state, otherwise store the data in the sensor attributes
+        if isinstance(ilo_data, (str, bytes, int, float)):
```

Comment:
Can you remove this logic and always assign it to the state. I know that it's pre-existing logic but it shouldn't have been merged